website/docs/r/vapp_firewall_rules.html.markdown

```diff
@@ -0,0 +1,119 @@
+    end_address   = "192.168.2.100"
+  }
+​
+}
+resource "vcd_vapp_firewall_rules" "vapp_fw" {
+  vapp_id        = vcd_vapp.TestAccVcdVAppForInsert.id
+  network_id     = vcd_vapp_network.vapp-net.id
+  default_action = "drop"
+
+  rule {
+    description      = "drop-ftp-out"
+    policy           = "drop"
+    protocol         = "tcp"
+    destination_port = "21"
+    destination_ip   = "any"
+    source_port      = "any"
+    source_ip        = "10.10.0.0/24"
+  }
+
+  rule {
+    description      = "allow-outbound"
+    policy           = "allow"
+    protocol         = "any"
+    destination_port = "any"
+    destination_ip   = "any"
+    source_port      = "any"
+    source_ip        = "10.10.0.0/24"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional; *v2.0+*) The name of organization to use, optional if defined at provider level. Useful when connected as sysadmin working across different organisations
+* `vdc` - (Optional; *v2.0+*) The name of VDC to use, optional if defined at provider level
+* `vapp_id` - (Required) The identifier of vApp
+* `network_id` - (Required) The identifier of vApp network
+* `default_action` - (Required) Either "allow" or "drop". Specifies what to do should none of the rules match
+* `log_default_action` - (Optional) Flag to enable logging for default action. Default value is false
+* `rule` - (Optional) Configures a firewall rule; see [Rules](#rules) below for details.
+
+<a id="rules"></a>
+## Rules
+
+Each firewall rule supports the following attributes:
+
+* `description` - (Optional) Name of the firewall rule
+* `enabled` - (Optional) 'true' value will enable firewall rule
+* `policy` - (Optional) Specifies what to do when this rule is matched. Either "allow" or "drop"
+* `protocol` - (Optional) The protocol to match. One of "tcp", "udp", "icmp", "any"" or "tcp&udp"
+* `destination_port` - (Optional) The destination port to match. Either a port number or "any"
+* `destination_ip` - (Optional) The destination IP to match. Either an IP address, IP range or "any"
+* `destination_vm_id` - (Optional) Destination VM identifier
+* `destination_vm_ip_type` - (Optional) The value can be one of: `assigned` - assigned internal IP be automatically choosen. `NAT`: NATed external IP will be automatically choosen
+* `destination_vm_nic_id` - (Optional) VM NIC ID to which this rule applies
+* `source_port` - (Optional) The source port to match. Either a port number or "any"
+* `source_ip` - (Optional) The source IP to match. Either an IP address, IP range or "any"
+* `source_vm_id` - (Optional) Source VM identifier
+* `source_vm_ip_type` - (Optional) The value can be one of: `assigned` - assigned internal IP be automatically choosen. `NAT`: NATed external IP will be automatically choosen
+* `source_vm_nic_id` - (Optional) VM NIC ID to which this rule applies
+* `enable_logging`- (Optional) 'true' value will enable rule logging. Default is false
```

Comment:
- Please add dots at the end of bullet point sentences above.
- Instead of quotes for constants (e.g. "allow", "tcp", "any") format them as code (i.e. `allow`, `tcp`, `any`).
- Format boolean values (i.e. `false`) as code.

Reply:
done